index.ts

```diff
@@ -0,0 +1,312 @@
+
+    const completeButton: HTMLButtonElement = document.createElement("button");
+    completeButton.classList.add(CSSClassName.CompleteButton);
+    completeButton.innerHTML = task.completed
+      ? incompleteIcon()
+      : completeIcon();
+    completeButton.addEventListener("click", () => this.completeTask(task.id));
+
+    const editButton = document.createElement("button");
+    editButton.classList.add(CSSClassName.EditButton);
+    editButton.innerHTML = editIcon();
+    editButton.addEventListener("click", () => this.editTask(task.id));
+
+    const deleteButton = document.createElement("button");
+    deleteButton.classList.add(CSSClassName.DeleteButton);
+    deleteButton.innerHTML = deleteIcon();
+    deleteButton.addEventListener("click", () => this.deleteTask(task.id));
+
+    taskBtns.appendChild(completeButton);
+    taskBtns.appendChild(editButton);
+    taskBtns.appendChild(deleteButton);
+
+    taskItem.appendChild(taskBody);
+    taskItem.appendChild(taskBtns);
+
+    let inserted: boolean = false;
+    for (let child of tasksList.children) {
+      if (child.id > taskItem.id) {
+        tasksList.insertBefore(taskItem, child);
+        inserted = true;
+        break;
+      }
+    }
+    if (!inserted) tasksList.appendChild(taskItem);
+  }
+  deleteTask(id: string): void {
+    this.tasks = this.tasks.filter((task) => task.id !== id);
+    this.displayTasks();
+  }
+  completeTask(id: string): void {
+    const taskIndex = this.tasks.findIndex((task) => task.id === id);
+    this.tasks[taskIndex].completed = !this.tasks[taskIndex].completed;
+    this.displayTasks();
+  }
+  editTask(id: string): void {
+    const task: Task | undefined = this.tasks.find((task) => task.id === id);
+    if (!task) return;
+    const taskItem = document.getElementById(id) as HTMLDivElement | null;
+    if (!taskItem) return;
+    taskItem.innerHTML = "";
+
+    const taskBody: HTMLDivElement = document.createElement("div");
+    taskBody.classList.add(CSSClassName.TaskBody);
+
+    const taskInfo: HTMLDivElement = document.createElement("div");
+    taskInfo.classList.add(CSSClassName.TaskInfo);
+
+    const editTaskTitle: HTMLInputElement = document.createElement("input");
+    editTaskTitle.classList.add(CSSClassName.TaskTitle);
+    editTaskTitle.value = task.title;
+    taskInfo.appendChild(editTaskTitle);
+
+    const editTaskDescription: HTMLInputElement =
+      document.createElement("input");
+    editTaskDescription.classList.add(CSSClassName.TaskDescription);
+    editTaskDescription.value = task.description;
+    taskInfo.appendChild(editTaskDescription);
+
+    taskBody.appendChild(taskInfo);
+
+    const taskBtns: HTMLDivElement = document.createElement("div");
+    taskBtns.classList.add(CSSClassName.TaskBtns);
+
+    const editPriority: HTMLSelectElement = document.createElement("select");
+    editPriority.classList.add(CSSClassName.EditTaskPriority);
+
+    const priorities: PriorityType[] = ["low", "medium", "high"];
```

Comment:
use the enum values and not static ones , I think you can sue something like `Object.values(PriorityType)`